src/relayFeeCalculator/relayFeeCalculator.ts

```diff
@@ -194,9 +195,29 @@ export class RelayFeeCalculator {
     return this.queries.getTokenPrice(tokenSymbol);
   }
 
+  /**
+   * Calculate the gas fee as a % of the amount to relay.
+   * @param deposit A valid deposit object to reason about
+   * @param amountToRelay The amount that we should fill the deposit for
+   * @param simulateZeroFill Whether to simulate a zero fill for the gas cost simulation
+   *        A fill of 1 wei which would result in a slow/partial fill.
+   *        You should do this if you're not worried about simulating a proper fill of a deposit
+   *        with a message or if you are worried a fill amount that could exceed the balance of
+   *        the relayer.
+   * @param relayerAddress The relayer that will be used for the gas cost simulation
+   * @param _tokenPrice The token price for normalizing fees
+   * @returns The fee as a % of the amount to relay.
+   * @note Setting simulateZeroFill to true will result on the gas costs being estimated
+   *       on a zero fill. However, the percentage will be returned as a percentage of the
+   *       amount to relay. This is useful for determining the maximum gas fee % that a
+   *       relayer may need to make on a regular fill. You will get differing results if
+   *       a message & recipient contract is provided as this function may not simulate with
+   *       the correct parameters to see a full fill.
+   */
   async gasFeePercent(
     deposit: Deposit,
     amountToRelay: BigNumberish,
+    simulateZeroFill = true,
```

Comment:
As mentioned offline, we should rename this to indicate its a 1 wei fill and also simulate a full fill instead of a partial

Reply:
fwiw, just for alignment, "zero fill" is the term used in the relayer to indicate a 1 wei fill (counter-intuitive). As I understand it, zero-fill also means 1 wei in this context.

Reply:
@nicholaspai this is using the `relayer-v2` terminology of a `zeroFill` being `1wei`. Setting `simulateZeroFill = false` will attempt to `fillRelay( )` with `amountToRelay` as the `MaxTokensToSend`.

Reply:
Should we add an additional flag to override this so that `amountToRelay` is ignored for the simulation purpose?

Reply:
oki'm on board with naming it `zeroFill` and no I don't think another flag is the correct way to do this. 

Why not do:

- if `simulateZeroFill` is `true`:
- pass in `{ ...deposit, amount: 1 wei }` when simulating the gas costs to simulate the full fill and then you can still derive `gasFeePercent` from `gasCost / amountToRelay`?

Reply:
I think I'd be okay with this. 

As an alternative, per @mrice32 we can usually assume that if a message is non-empty, then we should attempt to make a full fill, but if the message is not empty, we can default back to the original usage and attempt to fill with one unit. 

In the above case, we could also fix deposit.amount = 1. 

It's a shame that we moved the check against partial fills outside of the SDK because that would be a good decider of whether or not we should reject if the relayer balance is not enough to full fill a message.

TLDR: THoughts on removing `simulateZeroFill` in lieu of checking if `isMessageEmpty(deposit)` as the conditional for whether or not we attempt to simulate with a zero fill?

Reply:
I like having an explicit flag, personally. I don't love the idea of inferring based on other variables. We should be consistent on deciding this outside of this function or inside, and we've decided that outside is the way to go.

Secondly, 1 wei fills will probably not give an accurate estimation. They can round to 0, causing the math to change and no tokens to be transferred. I would suggest 100 units (small no matter which token) or similar. There _may not be_ rounding if the fees are all 0 with 1 wei, but I'm not 100% sure (we could test this). Thoughts on either testing this or making it 100 units instead of 1?

Reply:
Resolved offline with @mrice32